core/variant_parser.cpp

```diff
@@ -296,7 +296,10 @@ Error VariantParser::get_token(Stream *p_stream, Token &r_token, int &line, Stri
 				if (cchar == '-' || (cchar >= '0' && cchar <= '9')) {
 					//a number
 
-					String num;
+					const int num_buffer_size = 64;
+					CharType num_buffer[num_buffer_size];
```

Comment:
`num_buffer_size` needs to be a `static const int` in order to comply to the C++ standard. The way it is now is probably accepted by most compilers because they can detect it's a constant expression, but by the standard this is not a valid declaration.

With `-std=gnu++03` this might work, but it's again not C++ standard conform.

I mean it will most likely work on all compilers currently used, just pointing it out here. 😄